community/kernel/src/main/java/org/neo4j/kernel/impl/store/id/IdRangeIterator.java

```diff
@@ -71,12 +76,44 @@ public long next()
         }
     }
 
+    @Override
+    public IdRange nextIdBatch( int size )
+    {
+        int sizeLeft = size;
+        long[] rangeDefrag = EMPTY_LONG_ARRAY;
+        if ( position < defrag.length )
+        {
+            // There are defragged ids to grab
+            int numberOfDefrags = min( sizeLeft, defrag.length - position );
+            rangeDefrag = Arrays.copyOfRange( defrag, position, numberOfDefrags + position );
+            position += numberOfDefrags;
+            sizeLeft -= numberOfDefrags;
+        }
+
+        long rangeStart = 0;
+        int rangeLength = 0;
+        int rangeOffset = currentRangeOffset();
+        int rangeAvailable = length - rangeOffset;
+        if ( sizeLeft > 0 && rangeAvailable > 0 )
+        {
+            rangeStart = start + rangeOffset;
```

Comment:
why do we do`+ rangeOffset` part here? isn't range start/length independent from defrag ids and we can just start from `start` here instead?

Reply:
Do you see a case not covered by `IdRangeIteratorTest`?